src/main/java/javaslang/collection/IndexedSeq.java

```diff
@@ -346,3 +351,66 @@ default boolean startsWith(Iterable<? extends T> that, int offset) {
     IndexedSeq<Tuple2<T, Integer>> zipWithIndex();
 
 }
+
+interface IndexedSeqModule {
+    class LastIndexOfSlice {
+        static <T> int lastIndexOfSlice(IndexedSeq<T> t, IndexedSeq<T> slice, int end) {
+            if (end < 0) {
+                return -1;
+            }
+            if (t.isEmpty()) {
+                return slice.isEmpty() ? 0 : -1;
+            }
+            if (slice.isEmpty()) {
+                int len = t.length();
+                return len < end ? len : end;
+            }
+            int p = 0;
+            int result = -1;
+            while (t.length() >= p + slice.length()) {
+                int r = findSlice(t, p, slice);
+                if (r < 0) {
+                    return result;
+                }
+                if (r <= end) {
+                    result = r;
+                    p = r + 1;
+                } else {
+                    return result;
+                }
+            }
+            return result;
+        }
+
+        private static <T> int findSlice(IndexedSeq<T> t, int p, IndexedSeq<T> slice) {
+            if (t.isEmpty()) {
+                return slice.isEmpty() ? 0 : -1;
+            }
```

Comment:
we do not need to check this whole if - we've already done it in lastIndexOfSlice, right?